src/apps/logging/logging.cpp

```diff
@@ -257,6 +259,81 @@ namespace loggingapp
       };
       // SNIPPET_END: log_record_text
 
+      auto& historical_state = context.get_historical_state();
```

Comment:
This changes the pattern of how the handlers are created, instead of taking this it now takes this and historical_state.

Why not make historical_state part of this? I think it has the same life span

Reply:
Good point - we could store `historical_state` as a member since its the same lifespan as the tables. Alternatively it might make sense to pass this directly to the lambda (since historical queries likely need to have a distinct signature anyway), ie:

```
install("get_historical", [this](ccf::RequestArgs& args, AbstractStateCache& historical_state){ ... });
```

As below, I'd like to leave this API cleanup for a separate PR.